### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by viktarius**

### CODE REVIEW

File: postgres/src/routes/user.route.spec.ts [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    const updateUser = jest.fn((id, data) => Promise.resolve([data]));
+    const deleteUser = jest.fn((id) => Promise.resolve('ok'));
+    const addUserToGroup = jest.fn((u_id, g_id) => Promise.resolve('ok'));
+    return {getAllUsers, getUserById, addUser, updateUser, deleteUser, addUserToGroup};
+});
+
+let server: Server;
+let agent: supertest.SuperTest<supertest.Test>;
+
+beforeEach((done) => {
```
Comment (AlexeyBarkow) [inline postgres/src/routes/user.route.spec.ts:47]: before each and after each statements are usually put inside `describe`

File: postgres/src/routes/user.route.spec.ts [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    server = app.listen(4000, err => {
+        if (err) return done(err);
+
+        done();
+    });
+    agent = supertest(server);
```
Comment (AlexeyBarkow) [inline postgres/src/routes/user.route.spec.ts:53]: why not just `agent  = supertest(app)`?

Comment (AlexeyBarkow) [reply postgres/src/routes/user.route.spec.ts:53]: in this case you won't need to create server

File: postgres/src/routes/group.route.spec.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    const addGroup = jest.fn((data) => Promise.resolve([data]));
+    const updateGroup = jest.fn((id, data) => Promise.resolve([data]));
+    const deleteGroup = jest.fn((id) => Promise.resolve('ok'));
+    return {getAllGroups, getGroupById, addGroup, updateGroup, deleteGroup};
+});
+
+let server: Server;
+let agent: supertest.SuperTest<supertest.Test>;
+
+beforeEach((done) => {
```
Comment (AlexeyBarkow) [inline postgres/src/routes/group.route.spec.ts:41]: same comments as for users

File: postgres/src/routes/user.route.spec.ts (hunk 2 above)
Comment (viktarius) [reply postgres/src/routes/user.route.spec.ts:53]: I faced this issue: https://blog.campvanilla.com/jest-expressjs-and-the-eaddrinuse-error-bac39356c33a
and repeated this implementation 
https://www.gitmemory.com/issue/visionmedia/supertest/568/477130524

**Review by AlexeyBarkow — APPROVED** [review]

**Pull request merged**